metals/src/main/scala/scala/meta/internal/builds/SbtBuildTool.scala

```diff
@@ -83,42 +83,36 @@ case class SbtBuildTool(
     Files.deleteIfExists(metalsFile.toNIO)
   }
 
-  private def writeSbtMetalsPlugin(
+  private def writeSbtMetalsPlugins(
       workspace: AbsolutePath
   ): Unit = {
 
     def sbtMetaDirs(
-        dir: AbsolutePath,
-        parentHasSbtSrc: Boolean,
-        acc: List[AbsolutePath]
-    ): List[AbsolutePath] = {
-
-      val next = dir.resolve("project")
-      if (dir.exists) {
-        val files = dir.list.toList
-        val hasScalaSrc = files.exists(_.isScala)
+        meta: AbsolutePath,
+        acc: Set[AbsolutePath]
+    ): Set[AbsolutePath] = {
+      if (meta.exists) {
+        val files = meta.list.toList
         val hasSbtSrc = files.exists(f => f.isSbt && f.filename != "metals.sbt")
-        val goNext = hasScalaSrc || hasSbtSrc || parentHasSbtSrc
-
-        if (goNext)
-          sbtMetaDirs(next, hasSbtSrc, dir :: acc)
-        else
+        if (hasSbtSrc) {
```

Comment:
I think it should check the existence for `.scala` files too, not only `.sbt`.

However, it's probably it's hard to find a project that has sbt-scala sources somewhere deeper than in `.project/` directory

Reply:
We are not really registering those files within the correct build target, I will fix up together later.